### PR TITLE
Add ISSUE_TEMPLATE and PULL_REQUEST_TEMPLATE

### PR DESCRIPTION
**Pull request by Bowen12992**
Add `ISSUE_TEMPLATE` and `PULL_REQUEST_TEMPLATE`

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - Refactor (non-functional code changes)
+    - Documentation update (improvements or additions to documentation)
+    - Other (please describe)
+ -->
+
+### Description
+
+<!-- Briefly describe the changes made and the purpose of the changes.-->
+
+### Related Issues
```
Comment (tongxin) [inline .github/PULL_REQUEST_TEMPLATE.md]: Related Issues -> Issue

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+### Type of Change
+<!-- Select the type of change that applies:
+    - Bug fix (fixing an unintended behavior)
+    - New feature (introducing new functionality)
+    - Performance optimization (improving the efficiency of existing functionality)
+    - Refactor (non-functional code changes)
+    - Documentation update (improvements or additions to documentation)
+    - Other (please describe)
+ -->
```
Comment (tongxin) [inline .github/PULL_REQUEST_TEMPLATE.md]: Let's remove this section for simplicity.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+List any related issues that this PR resolves, if applicable, for example:
+- Resolves #123
+- Associated with Feature #456
+-->
+
+### Additional Information
+
+<!--
+Provide any other relevant information, such as design decisions, risk assessments, or areas that require special attention.
+-->
```
Comment (tongxin) [inline .github/PULL_REQUEST_TEMPLATE.md]: Remove this.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+### Checklist
```
Comment (tongxin) [inline .github/PULL_REQUEST_TEMPLATE.md]: Change this to `Progress`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+- [ ] Code has been self-reviewed and tested.
+- [ ] Testing can cover more than 90% of the new code.
```
Comment (tongxin) [inline .github/PULL_REQUEST_TEMPLATE.md]: The checklist should not duplicate bot triggered checks.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+- [ ] Documentation has been updated (if applicable).
+- [ ] Code review approval has been obtained.
+- [ ] Added benchmark tests for the new operation.
+
+### Performance Results
```
Comment (tongxin) [inline .github/PULL_REQUEST_TEMPLATE.md]: Performance Results => Performance

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+### Additional Information
+
+<!--
+Provide any other relevant information, such as design decisions, risk assessments, or areas that require special attention.
+-->
+
+### Checklist
+
+- [ ] Code has been self-reviewed and tested.
```
Comment (tongxin) [inline .github/PULL_REQUEST_TEMPLATE.md]: - Change is properly reviewed (1 required, 2 recommended).
- Change is responded to an issue.
- Change is fully covered by a UT.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Bowen12992) [reply .github/PULL_REQUEST_TEMPLATE.md]: ok

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (Bowen12992) [reply .github/PULL_REQUEST_TEMPLATE.md]: OK

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+<!--
+List any related issues that this PR resolves, if applicable, for example:
+- Resolves #123
+- Associated with Feature #456
+-->
+
+### Progress
+
+- [ ] Change is properly reviewed (1 required, 2 recommended).
```
Comment (tongxin) [inline .github/PULL_REQUEST_TEMPLATE.md]: 1 reviewer required

**Review by tongxin — DISMISSED** [review]
LGTM

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 8 above)
Comment (Bowen12992) [reply .github/PULL_REQUEST_TEMPLATE.md]: done

**Review by tongxin — APPROVED** [review]
LGTM

**Pull request merged**